graylog2-web-interface/src/components/search/MessageFieldExtractorActions.jsx

```diff
@@ -25,17 +25,34 @@ const MessageFieldExtractorActions = React.createClass({
     );
   },
   render() {
-    return (
-      <div className="message-field-actions pull-right">
-        <DropdownButton pullRight
-                     bsSize="xsmall"
-                     title="Select extractor type"
-                     key={1}
-                     id={`select-extractor-type-dropdown-field-${this.props.fieldName}`}>
-          {ExtractorUtils.EXTRACTOR_TYPES.map(extractorType => this._formatExtractorMenuItem(extractorType))}
-        </DropdownButton>
-      </div>
-    );
+    let messageField = this.props.message.fields[this.props.fieldName];
+    if(typeof messageField === 'string') {
+      return (
+          <div className="message-field-actions pull-right">
+            <DropdownButton pullRight
+                            bsSize="xsmall"
+                            title="Select extractor type"
+                            key={1}
+                            id={`select-extractor-type-dropdown-field-${this.props.fieldName}`}>
+                {ExtractorUtils.EXTRACTOR_TYPES.map(extractorType => this._formatExtractorMenuItem(extractorType))}
+            </DropdownButton>
+          </div>
+        );
+    } else {
+        return (
+            <div className="message-field-actions pull-right">
+              <DropdownButton pullRight
+                              bsSize="xsmall"
+                              title="Select extractor type"
+                              key={1}
+                              id={`select-extractor-type-dropdown-field-${this.props.fieldName}`}>
+                <MenuItem key={`menu-item-disabled`} disabled={true}>
```

Comment:
Here another couple of nitpicks: 

- `key` can be a regular string
- We don't need to pass the boolean value to the `disabled` flag (it breaks the [react/jsx-boolean-value](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-boolean-value.md) rule)